src/main/java/me/botsko/prism/serializers/items/ItemStackSerializer.java

```diff
@@ -0,0 +1,295 @@
+                final String[] enchArgs = ench.split(":");
+                Enchantment enchantment = Enchantment.getByKey(NamespacedKey.minecraft(enchArgs[0]));
+
+                // Restore book enchantment
+                if (enchantment != null) {
+                    if (item.getType() == Material.ENCHANTED_BOOK) {
+                        final EnchantmentStorageMeta bookEnchantments = (EnchantmentStorageMeta) item.getItemMeta();
+                        if (bookEnchantments != null) {
+                            bookEnchantments.addStoredEnchant(enchantment, Integer.parseInt(enchArgs[1]), false);
+                            item.setItemMeta(bookEnchantments);
+                        }
+                    } else {
+                        item.addUnsafeEnchantment(enchantment, Integer.parseInt(enchArgs[1]));
+                    }
+                }
+            }
+        }
+
+        ItemMeta meta = item.getItemMeta();
+        if (name != null) {
+            if (meta != null) {
+                meta.setDisplayName(name);
+            }
+        }
+        if (lore != null) {
+            if (meta != null) {
+                meta.setLore(Arrays.asList(lore));
+            }
+        }
+        // Leather color
+        if (meta instanceof LeatherArmorMeta && color > 0) {
+            final LeatherArmorMeta lam = (LeatherArmorMeta) meta;
+            lam.setColor(Color.fromRGB(color));
+            item.setItemMeta(lam);
+        } else if (meta instanceof SkullMeta && owner != null) {
+            final SkullMeta skull = (SkullMeta) meta;
+            skull.setOwningPlayer(Bukkit.getOfflinePlayer(EntityUtils.uuidOf(owner)));
+            item.setItemMeta(skull);
+        } else if (meta instanceof BookMeta) {
+            final BookMeta bookMeta = (BookMeta) meta;
+            bookMeta.setAuthor(by);
+            bookMeta.setTitle(title);
+            bookMeta.setPages(content);
+            item.setItemMeta(bookMeta);
+        } else if (meta instanceof FireworkEffectMeta && effectColors != null
+                && effectColors.length > 0) {
+            return deserializeFireWorksMeta(item, meta);
+        } else if (meta instanceof BannerMeta && bannerMeta != null) {
+            Map<String, String> stringStringMap = bannerMeta;
+            List<Pattern> patterns = new ArrayList<>();
+            stringStringMap.forEach((patternIdentifier, dyeName) -> {
+                PatternType type = PatternType.getByIdentifier(patternIdentifier);
+                try {
+                    DyeColor color = DyeColor.valueOf(dyeName);
+                    if (type != null) {
+                        Pattern p = new Pattern(color, type);
+                        patterns.add(p);
+                    }
+                } catch (IllegalArgumentException e){
+                    Prism.debug("Found illegal DyeColor stored in serialized banner: "+e.getMessage());
+                    Prism.debug("Data:  " +bannerMeta.toString());
```

Comment:
Codacy found an issue: [WhitespaceAround: '+' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=6368169)